### PR TITLE
XSS bug fix

### PR DESCRIPTION
**Pull request by PraveenCTzen**
JIRA Issue: [WEB-2516](https://wizrocket.atlassian.net/browse/WEB-2516)
​

**Background**
We've identified a critical defect in the web popup's custom HTML feature, allowing malicious script injection. To mitigate this risk, we've introduced iframe sandboxing for enhanced security against cross-site scripting (XSS) vulnerabilities.

**Implementation**
To address the risk of cross-site scripting (XSS) vulnerabilities, we've implemented iframe sandboxing for enhanced security
​

**Testing Strategy**
We've extensively tested the solution across various custom-html web popups, examining notification viewing and clicking events, as well as CleverTap actions such as 'event.push,' 'profile.push,' and 'onUserLogin.push.' Additionally, we're using a dashboard flag to distinguish between old and new campaigns, applying the fix solely to new ones, while ensuring it doesn't interfere with form-based templates.

**Impact Areas**
The change will have a big impact because we're sandboxing the iframe. This might cause problems for features that rely on the iframe talking to the main window. Before, users would send events or update profiles using 'window.parent' commands. Now, in new campaigns, they'll need to remove 'window.parent' from these commands because of the sandboxing. For example, they'll switch from 'window.parent.clevertap.event.push(eventName)' to just 'clevertap.event.push(eventName).' This makes sure everything still works with the sandboxed iframe

[WEB-2516]: https://wizrocket.atlassian.net/browse/WEB-2516?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/util/tr.js [hunk 1]
```diff
@@ -496,75 +498,151 @@ const _tr = (msg, {
...
+  }
+
+  const ctEventhandler = (html) => {
+    const ctScript = `
+     var clevertap = {
+      event: {
+        push: (eventName) => {
+          window.parent.postMessage({
+            action: 'Event',
+            value: eventName
```
Comment (kkyusuftk) [inline src/util/tr.js:573]: can we have this script in a different file? 
Will reduce the number of lines in this file

File: src/util/tr.js [hunk 2]
```diff
@@ -815,22 +912,44 @@ const _tr = (msg, {
...
+          if (event?.data?.action === 'Event') {
+            window.clevertap.event.push(event.data.value)
+          } else if (event?.data?.action === 'Profile') {
+            window.clevertap.profile.push(event.data.value)
+          } else if (event?.data?.action === 'OUL') {
+            window.clevertap.onUserLogin.push(event.data.value)
+          } else if (event?.data?.action === 'closeInterstitialPopUp') {
+            var interstitialWrapper = window.document.getElementById('intentPreview')
+            var interstitialOverlay = window.document.getElementById('intentOpacityDiv')
+            setTimeout(() => {
```
Comment (kkyusuftk) [inline src/util/tr.js:941]: same comments as above. hard coded strings and switch statements

**Review by kkyusuftk — CHANGES_REQUESTED** [review]

File: src/util/tr.js [hunk 3]
```diff
@@ -496,75 +498,151 @@ const _tr = (msg, {
...
+          } else if (event?.data?.action === 'OUL') {
+            window.clevertap.onUserLogin.push(event.data.value)
+          } else if (event?.data?.action === 'closeBoxPopUp') {
+            var boxWrapper = window.document.getElementById('wizParDiv0')
+            setTimeout(() => { boxWrapper.remove() }, 0)
+          } else if (event?.data?.action === 'closeBannerPopUp') {
+            var bannerWrapper = window.document.getElementById('wizParDiv2')
+            setTimeout(() => { bannerWrapper.remove() }, 0)
+          }
+        })
```
[the remark this answers is not in the document]
Comment (PraveenCTzen) [reply src/util/tr.js]: Pushed the updated code

File: src/util/tr.js (hunk 2 above)
Comment (PraveenCTzen) [reply src/util/tr.js]: Done!
Pushed the updated code!

File: src/util/tr.js [hunk 4]
```diff
@@ -496,75 +499,130 @@ const _tr = (msg, {
...
+          }, 0)
+          break
+        case ActionType.CLOSE_BANNER_POPUP:
+          setTimeout(() => {
+            bannerWrapper = window.document.getElementById('wizParDiv2')
+            bannerWrapper && bannerWrapper.remove()
+          }, 0)
+          break
+        default:
+          // Handle unknown action
```
Comment (kkyusuftk) [inline src/util/tr.js:569]: lets have a log here. We need to know whether someone is passing some unknown action

File: src/util/tr.js [hunk 5]
```diff
@@ -496,75 +498,151 @@ const _tr = (msg, {
...
+        })
+        contentDiv = ''
+      } else {
+        // adjust iframe and body height of html inside correctly
+        contentHeight = document.getElementById('wiz-iframe').contentDocument.getElementById('contentDiv').scrollHeight
+        if (displayObj['custom-editor'] !== true && !isBanner) {
+          contentHeight += 25
+        }
+        document.getElementById('wiz-iframe').contentDocument.body.style.margin = '0px'
+        document.getElementById('wiz-iframe').style.height = contentHeight + 'px'
```
[the remark this answers is not in the document]
Comment (kkyusuftk) [reply src/util/tr.js:588]: no hardcoded `wiz-iframe`

File: src/util/tr.js [hunk 6]
```diff
@@ -815,22 +892,59 @@ const _tr = (msg, {
...
+    iframe.srcdoc = html
+
+    let contentDiv
+    const ActionType = {
+      GET_NOTIFICATION_DATA: 'getnotifData',
+      EVENT: 'Event',
+      PROFILE: 'Profile',
+      OUL: 'OUL',
+      CLOSE_INTERSTITIAL_POPUP: 'closeInterstitialPopUp'
+    }
```
Comment (kkyusuftk) [inline src/util/tr.js:912]: would prefer to have this in a different file

File: src/util/tr.js [hunk 7]
```diff
@@ -496,75 +499,130 @@ const _tr = (msg, {
...
+
+    const ActionType = {
+      ADJUST_IFRAME_HEIGHT: 'adjustIFrameHeight',
+      UPDATE_HEIGHT: 'update height',
+      GET_NOTIFICATION: 'getnotif',
+      EVENT: 'Event',
+      PROFILE: 'Profile',
+      OUL: 'OUL',
+      CLOSE_BOX_POPUP: 'closeBoxPopUp',
+      CLOSE_BANNER_POPUP: 'closeBannerPopUp'
```
Comment (kkyusuftk) [inline src/util/tr.js:530]: same, can you shift this to a different constants file

**Review by kkyusuftk — CHANGES_REQUESTED** [review]

File: src/util/tr.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (PraveenCTzen) [reply src/util/tr.js]: Done

File: src/util/tr.js (hunk 7 above)
Comment (PraveenCTzen) [reply src/util/tr.js]: Updated the code

**Pull request merged**